File: solutions/2107-find-unique-binary-string/solution.js
/**
 * @param {string[]} nums
 * @return {string}
 */
var findDifferentBinaryString = function(nums) {
    const n = nums.length;
    const s = new Set(nums);

    function randomBin(){
        const out = []
        for(let i = 0; i < n; i++){
            if(Math.random() > 0.5){
                out.push(1)
            }else{
                out.push(0)
            }
        }
        return out.join("");
    }

    let random = randomBin();
    while(s.has(random)){
        random = randomBin();
    }

    return random;
};

File: solutions/2107-find-unique-binary-string/solution.py
class Solution:
    def findDifferentBinaryString(self, nums: List[str]) -> str:
        s = set()
        for num in nums:
            s.add(int(num,2))

        for i in range(2 ** len(nums)):
            print(i)
            if i not in s:
                out = bin(i)[2:]
                out = "0" * (len(nums) - len(out)) + out
                return out
